### PR TITLE
:sparkles: Added a button to refresh dataset list when tiled server i…

### PR DESCRIPTION
**Pull request by zhuowenzhao**
Changelog:
1. encapsulated data util functions as a class object, which is shared across scripts after instantiated.
2. added a refresh button next to the dataset dropdown to update the available datasets after tiled server is updated (re-indexed).

Note: removing a dataset from tiled will trigger missing data error if the dropdown selection was the one that is removed. Adding new dataset to tiled server should be fine.

### CODE REVIEW

File: utils/data_utils.py [hunk 1]
```diff
@@ -16,130 +16,146 @@
...
-        ar = img[int(idx)]
-        raw.append(ar)
-    try:
-        raw = np.stack(raw)
-        mask = np.stack(mask)
-    except ValueError:
-        return "No annotations to process."
-
-    return
+
```
Comment (dylanmcreynolds) [inline utils/data_utils.py:19]: This is a little picky, but cab we call this class something that links it to tiled, like maybe `TiledDataLoader`

File: utils/data_utils.py [hunk 2]
```diff
@@ -16,130 +16,146 @@
...
+            for line in f:
+                if line.strip():
+                    json_data = json.loads(line)
+                    json_data["data"] = json.loads(json_data["data"])
+                    annotated_slices = list(json_data["data"][0]["annotations"].keys())
+        return annotated_slices
+
+    
+    @staticmethod
+    def DEV_load_exported_json_data(file_path, USER_NAME, PROJECT_NAME):
```
Comment (dylanmcreynolds) [inline utils/data_utils.py:96]: What's the need for the `DEV` name?

File: utils/data_utils.py [hunk 3]
```diff
@@ -16,130 +16,146 @@
...
+            if data["user"] == USER_NAME and data["source"] == PROJECT_NAME
+        ]
+        if data:
+            data = sorted(data, key=lambda x: x["time"], reverse=True)
+
+        return data
+
+    
+    @staticmethod
+    def DEV_filter_json_data_by_timestamp(data, timestamp):
```
Comment (dylanmcreynolds) [inline utils/data_utils.py:126]: Same comment about `Dev`

**Review by dylanmcreynolds — COMMENTED** [review]
I had a few comments about naming.

File: utils/data_utils.py (hunk 2 above)
Comment (Wiebke) [reply utils/data_utils.py:96]: Most functions that were implemented as a placeholder while some of the connections to Tiled/Computing API/... were missing were marked with a `DEV_` prefix.

File: utils/data_utils.py (hunk 1 above)
Comment (Wiebke) [reply utils/data_utils.py:19]: I like that!

**Review by Wiebke — CHANGES_REQUESTED** [review]
I like the refactoring of the data loading functionality into a class, but I think we should differentiate between what data is being acted upon. 
Saving and loading annotations has separate common variables (currently the filename, in the future the `MASK_TILED_URI`) from the raw data and I would thus suggest to place these in a separate class. 

I could take this up when working on the refactoring of the annotations loading and saving functionality. For now, renaming `DataUtils` as Dylan suggested would be great.

Could you additionally reformat the code? We previously had Github actions with checks for formatting, linting, and testing but these were unintentionally removed in #158. 
 (`pip install -r requirements-dev.txt` and then `black .`)?

File: utils/data_utils.py (hunk 1 above)
Comment (zhuowenzhao) [reply utils/data_utils.py:19]: I like this idea too. Now that I am thinking about it, I think it aligns with what Tanny or Wiebke was thinking about implementing a data loader in the future.

**Comment by zhuowenzhao** [comment]
> I like the refactoring of the data loading functionality into a class, but I think we should differentiate between what data is being acted upon. Saving and loading annotations has separate common variables (currently the filename, in the future the `MASK_TILED_URI`) from the raw data and I would thus suggest to place these in a separate class.
> 
> I could take this up when working on the refactoring of the annotations loading and saving functionality. For now, renaming `DataUtils` as Dylan suggested would be great.
> 
> Could you additionally reformat the code? We previously had Github actions with checks for formatting, linting, and testing but these were unintentionally removed in #158. (`pip install -r requirements-dev.txt` and then `black .`)?

I agree with the different variables are needed for loading and saving Tiled datasets. But there is also a possibility to add additional args for the saving function in TiledDataLoader class. I guess this can be further discussed in the future. 

I have renamed the class and object. Please have a check :)

**Review by Wiebke — APPROVED** [review]
Thanks, This is great!

Deferring the formatting discussion to later, going ahead with merging.

**Pull request merged**